components/dashboard/src/usage/download/DownloadUsage.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, useCallback, useState } from "react";
+import { Button } from "../../components/Button";
+import { downloadUsageCSV } from "./download-usage-csv";
+import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
+import { Dayjs } from "dayjs";
+import { useToast } from "../../components/toasts/Toasts";
+import { useCurrentOrg } from "../../data/organizations/orgs-query";
+
+type Props = {
+    attributionId: AttributionId;
+    startDate: Dayjs;
+    endDate: Dayjs;
+};
+export const DownloadUsage: FC<Props> = ({ attributionId, startDate, endDate }) => {
+    const { data: org } = useCurrentOrg();
+    const [isSaving, setIsSaving] = useState(false);
+    const { toast } = useToast();
+
+    const handleDownload = useCallback(async () => {
+        if (!org) {
+            return;
+        }
+
+        setIsSaving(true);
+
+        try {
+            const downloaded = await downloadUsageCSV({
+                orgName: org?.slug ?? org?.id,
+                attributionId: AttributionId.render(attributionId),
+                from: startDate.startOf("day").valueOf(),
+                to: endDate.endOf("day").valueOf(),
+            });
+
+            if (!downloaded) {
+                toast("There are no usage records for that date range");
+            }
+        } catch (e) {
+            console.error(e);
+            toast(`There was a problem downloading your usage data: ${e.message}`);
+        }
+
+        setIsSaving(false);
+    }, [attributionId, endDate, org, startDate, toast]);
+
+    return (
+        <Button type="secondary" loading={isSaving} onClick={handleDownload}>
+            Download as CSV
```

Comment:
**question**: Instead of relying on the loading state of the button, what do you think of using a toast notification when the process to write the export file (CSV) takes more than a second? Too much for nothing?

| Toast notification when export takes more than X seconds |
|-|
| <img width="690" alt="toast" src="https://github.com/gitpod-io/gitpod/assets/120486/c4ae8d86-b412-4707-b083-a3cfa477a199"> |

Reply:
**thought**: I just saw your loom you posted earlier in a [relevant discussion](https://gitpod.slack.com/archives/C02EN94AEPL/p1685501382073799) (internal) and noticed you tried using a toast here!

I would not download the export and re-surface the link to re-download. I'd use a toast only to show progress of the process of writing the export file, and then either download the file or keep the export on our end until the user clicks the download button—if we go down the second route we could have an expiration time for the export, but maybe it's an overkill at the moment. 💭 

| Toasty after completing writing export file |
|-|
| ![Frame 1388](https://github.com/gitpod-io/gitpod/assets/120486/59212f25-8245-4bfe-b7ca-4a8704a75e01) |